frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.tsx

```diff
@@ -34,41 +34,20 @@ export const AccessControlTab = ({ org, app }: AccessControlTabProps): ReactNode
 
   const {
     status: appMetadataStatus,
-    isLoading: appMetadataLoading,
     data: appMetadata,
     error: appMetadataError,
   } = useAppMetadataQuery(org, app);
 
   const { mutate: updateAppMetadataMutation } = useAppMetadataMutation(org, app);
 
-  const [partyTypesAllowed, setPartyTypesAllowed] = useState<PartyTypesAllowed>(
-    appMetadata?.partyTypesAllowed ?? initialPartyTypes,
-  );
-
-  useEffect(() => {
-    if (!appMetadataLoading) {
-      setPartyTypesAllowed(appMetadata?.partyTypesAllowed ?? initialPartyTypes);
-    }
-  }, [appMetadataLoading, appMetadata]);
-
-  /**
-   * Update the selected party types when clicking an option
-   */
-  const handleChange = (partyTypes: string[]) => {
-    const newPartyTypesAllowed = { ...partyTypesAllowed };
+  const handleChange = (newPartyTypes: string[], currentPartyTypesAllowed: PartyTypesAllowed) => {
+    const newPartyTypesAllowed = { ...currentPartyTypesAllowed };
 
-    Object.keys(partyTypesAllowed).forEach((key) => {
-      newPartyTypesAllowed[key] = partyTypes.includes(key);
+    Object.keys(currentPartyTypesAllowed).forEach((key) => {
+      newPartyTypesAllowed[key] = newPartyTypes.includes(key);
     });
-
-    setPartyTypesAllowed(newPartyTypesAllowed);
-  };
-
-  /**
-   * Save the metadata with the new party types
-   */
-  const handleSavePartyTypes = () => {
-    updateAppMetadataMutation({ ...appMetadata, partyTypesAllowed });
+    console.log('newPartyTypesAllowed', newPartyTypesAllowed);
```

Comment:
I think you forgot to remove this one. 😊

Reply:
ooh, thanks! 😄